core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1939,24 +1941,23 @@ private[spark] class DAGScheduler(
       hostToUnregisterOutputs: Option[String],
       maybeEpoch: Option[Long] = None): Unit = {
     val currentEpoch = maybeEpoch.getOrElse(mapOutputTracker.getEpoch)
+    logDebug(s"Removing executor $execId, fileLost: $fileLost, currentEpoch: $currentEpoch")
```

Comment:
> See explanation in my response to your comment about the previous logDebug.
But more importantly, this logDebug tells you that the method is invoked. Then depending on whether there is additional log output, you know if one or both of the if clauses are executed as well. (Knowing the value of fileLost may rule out one of the clauses.)



What if the executor does not meet the if condition? In that case, the executor is not going to be removed.

Reply:
Ah, you're objecting to the *wording* of the debug message?
I'll change it from "Removing executor ..." to "Considering removal of executor ...".